server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionCoordinator.java

```diff
@@ -265,14 +264,11 @@ public void run() {
       throw new IllegalStateException("Exception getting Coordinator lock", e);
     }
 
-    try {
-      MetricsUtil.initializeMetrics(getContext().getConfiguration(), this.applicationName,
-          clientAddress, getContext().getInstanceName());
-    } catch (ClassNotFoundException | InstantiationException | IllegalAccessException
-        | IllegalArgumentException | InvocationTargetException | NoSuchMethodException
-        | SecurityException e1) {
-      LOG.error("Error initializing metrics, metrics will not be emitted.", e1);
-    }
+    MetricsInfo metricsInfo = getContext().getMetricsInfo();
+    metricsInfo.addServiceTags(getApplicationName(), clientAddress);
+
+    metricsInfo.addMetricsProducers();
```

Comment:
I don't think this does anything if you don't pass a parameter. Might want to add a check to `addMetricsProducers` that at least one parameter is passed.

Reply:
I'll remove the call in CompactionCoordinator. I'd like to leave an empty call as a no-op, but I will add logging.  It has no effect, so seems unnecessary to fail if called without providing a producer.

Reply:
Addressed in 601f167e80